File: 1261.py
import sys

m, n = map(int, sys.stdin.readline().split())

dict_words = {}

for _ in range(m):
    word, value = sys.stdin.readline().split()
    dict_words[word] = int(value)

for _ in range(n):
    total = 0
    while True:
        line = sys.stdin.readline().strip()
        if line == ".":
            break
        for word in line.split():
            total += dict_words.get(word, 0)
    print(total)

File: 1261.js
const fs = require("fs");

const input = fs.readFileSync(0, "utf8").trim().split("\n");
let idx = 0;

const [M, N] = input[idx++].split(" ").map(Number);

const dict = {};

for (let i = 0; i < M; i++) {
    const [word, value] = input[idx++].split(" ");
    dict[word] = Number(value);
}

for (let i = 0; i < N; i++) {
    let total = 0;
    while (true) {
        const line = input[idx++].trim();
        if (line === ".") break;

        for (const word of line.split(" ")) {
            total += dict[word] || 0;
        }
    }
    console.log(total);
}
